### DISCUSSION TITLE
How to make the traceids of producers and consumers consistent

### DISCUSSION THREAD

**Discussion by XTUxiongda** [apache/skywalking#8210] [Q&A] [answered]
I use rocketmq.
I hope that during link tracking, the traceids of producers and consumers in the whole process are consistent.
How can i do that?

**Comment by wu-sheng** [comment, marked as answer]
@HScarb I think you may be better to answer this.  

To @XTUxiongda RocketMQ plugin is a legacy from years ago, no plugin e2e testing provided. So, I really don't have anything to share. But generally, if you are not using MQ's typical listener mode(a callback), but you are using a fetcher like Kafka provided, you need a tracer wrapping the function of fetching message and continuing process.

A document about wrapping a method with tracing context, https://skywalking.apache.org/docs/skywalking-java/v8.8.0/en/setup/service-agent/java-agent/application-toolkit-trace/.

**Reply by XTUxiongda** [reply]
Maybe the question I really want to ask is：
When producers produce messages，I can use TraceContext.traceId() get the traceId.
i want put the traceid to the message, then send to the consumer.
but, i can not find any way to change the TraceId in the TraceContext

**Reply by wu-sheng** [reply]
No, no manual propagation for tracing context, including trace id.
We have done the thing you described.

**Reply by wu-sheng** [reply]
All things should be auto, but without test cases, I can't share what it looks like.

**Reply by XTUxiongda** [reply]
For example, when I call an interface of service A, then a traceid is generated. Service A will send a message to service B. I want to query the complete link on the page through this traceid. What should I do?Are there any ready-made solutions or suggestions?

**Reply by wu-sheng** [reply]
I even don't know how to continue, you seem having big gap of understanding SkyWalking Java agent.
Anyway, I will wait for RocketMQ people to answer. 
General, nothing should be done, installking agent should be enough.

**Reply by XTUxiongda** [reply]
OK, thank you for your patience. I really don't know enough about this. I'll study hard again. Thank you very much.
